sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -30,23 +30,83 @@
+    ///
+    /// Simple 'Noncompliant' comment. Will mark the current line as expecting the primary location of an issue.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant
+    /// </code>
+    ///
+    /// 'Secondary' location comment. Must be used together with a main primary location to mark the expected line of a
+    /// secondary location.
+    /// <code>
+    ///     if (myCondition) // Noncompliant
+    ///     {
+    ///       var a = null; // Secondary
+    ///     }
+    /// </code>
+    ///
+    /// Using offsets. Using @[+-][0-9]+ after a 'Noncompliant' or 'Secondary' comment will mark the expected location to be
+    /// offset by the given number of lines.
+    /// <code>
+    ///  private void MyMethod() // Noncompliant@+2 - issue is actually expected 2 lines after this comment
+    /// </code>
+    ///
+    /// Checking the issue message. The message raised by the issue can be checked using the {{expected message}} pattern.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant {{Remove this unused private method}}
+    /// </code>
+    ///
+    /// Checking the precise/exact location of an issue. Only one precise location or column location can be present
+    /// at one time. Precise location is used by adding '^^^^' comment under the location where the issue is expected.
+    /// The alternative column location pattern can be used by following the 'Noncompliant' or 'Secondary' comment
+    /// with '^X#Y' where 'X' is the expected start column and Y the length of the issue.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant
+    /// //  ^^^^^^^
+    /// </code>
+    /// <code>
+    ///     private void MyMethod() // Noncompliant ^4#7
+    /// </code>
+    /// 
+    /// Multiple issues per line. To declare that multiple issues are expected, each issue must be assigned an id. All
+    /// secondary locations associated with an issue must have the same id. Note that it is not possible to have multiple
+    /// precise/column locations on a single line.
+    /// <code>
+    ///     var a = null; // Noncompliant [myId2]
+    ///     if (myCondition) // Noncompliant [myId1, myId3]
+    ///     {
+    ///       a = null; // Secondary [myId1, myId2]
+    ///     }
+    /// </code>
+    ///
+    /// Note that most of the previous patterns can be used together when needed.
+    /// <code>
+    ///     private void MyMethod() // Noncompliant@+1 ^4#7 [MyIssueId] {{Remove this unused private method}}
+    /// </code>
+    /// <code>
+    /// </summary>
     public class IssueLocationCollector
     {
+        private const string EXACT_COLUMN_PATTERN = @"\s*(\^(?<columnStart>[0-9]+)#(?<length>[0-9]+))?";
         private const string OFFSET_PATTERN = @"(?:@(?<sign>[\+|-]?)(?<offset>[0-9]+))?";
         private const string ISSUE_IDS_PATTERN = @"\s*(\[(?<issueIds>.*)\])*";
         private const string MESSAGE_PATTERN = @"\s*(\{\{(?<message>.*)\}\})*";
```

Comment:
I don't think it was desired to allow for empty message, as it is optional anyways. I don't see any problem to allow `{` and `}` inside the message itself, as it could be valid to have such characters